### PR TITLE
chromium: update to 80.0.3987.100

### LINKED ISSUE #354: chromium: update to 80.0.3987.116
Stable release announcements since our latest update:
- https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop_18.html
- https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop_13.html
- https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop_11.html
- https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop.html
- https://chromereleases.googleblog.com/2020/01/stable-channel-update-for-desktop_16.html

Targets and platforms we need to check:

#### chromium-x11:
- [x] thud, clang, MACHINE="intel-core2-32"
- [x] thud, clang, MACHINE="intel-corei7-64"
- [x] warrior, clang, MACHINE="raspberrypi3"
- [x] warrior, clang, MACHINE="intel-corei7-64"

#### chromium-ozone-wayland:
- [x] thud, clang, MACHINE="m3ulcb"
- [x] thud, clang, MACHINE="qemuarm64"
- [x] warrior, clang, MACHINE="raspberrypi3"

### PR DESCRIPTION
**Pull request by msisov**
Stable release announcements since our latest update:
https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop_11.html
https://chromereleases.googleblog.com/2020/02/stable-channel-update-for-desktop.html
https://chromereleases.googleblog.com/2020/01/stable-channel-update-for-desktop_16.html

Changes:
1. Remove unbundle libxml as chromium limits xml gn visibility now
   for security reasons. See https://crrev.com/c/1894877
2. Backport some gcc fixes.

Known problems:
There is a known problem with chromium ozone/wayland video playback.
The patch will be backported as soon as it is merged to upstream.

Test-built with:
* chromium-x11
  * thud, MACHINE="intel-corei7-64"
  * thud, MACHINE="intel-core2-32"
  * warrior, MACHINE="intel-corei7-64"
  * warrior, MACHINE="intel-corei7-64"
* chromium-ozone-wayland:
  * thud, MACHINE="qemuarm64"
  * thud, MACHINE="m3ulcb"
  * warrior, MACHINE="raspberrypi3"

License changes
---------------
Removed licenses:
* third_party/flot/src/LICENSE.txt:
  <https://crrev.com/c/1914590>

Added/Changed licenses:
* third_party/boringssl/src/LICENSE:
  <https://crrev.com/c/1951086>
* third_party/feed/src/LICENSE:
  <https://crrev.com/c/1917083>
* third_party/mesa_headers/LICENSE:
  <https://crrev.com/c/1813353>
* third_party/pdfium/LICENSE:
  <https://pdfium-review.googlesource.com/c/pdfium/+/63170>
* third_party/skia/LICENSE:
  <https://skia-review.googlesource.com/c/skia/+/255305>
* third_party/android_deps/libs/androidx_concurrent_concurrent_futures/LICENSE
  third_party/android_deps/libs/com_google_guava_listenablefuture/LICENSE
  <https://crrev.com/c/1872642>
* third_party/android_deps/libs/com_google_guava_listenablefuture/LICENSE
  third_party/android_deps/libs/javax_inject_javax_inject/LICENSE
  <https://crrev.com/c/1183903>
* third_party/libgifcodec/LICENSE.md
  <https://crrev.com/c//1944286>
* third_party/wuffs/LICENSE
  <https://crrev.com/c/1859536>

Fixes #354
Signed-off-by: Maksim Sisov <msisov@igalia.com>

### CODE REVIEW

File: recipes-browser/chromium/chromium.inc [hunk 1]
```diff
@@ -202,7 +202,7 @@ LIC_FILES_CHKSUM = "\
     file://${S}/third_party/arcore-android-sdk/LICENSE;md5=c1bc4f7fbdd7a5a3f2c0dfefd457665e \
     file://${S}/third_party/ashmem/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/axe-core/LICENSE;md5=f27a50d2e878867827842f2c60e30bfc \
-    file://${S}/third_party/boringssl/src/LICENSE;md5=e91719b7727a2d3464ec006154275572 \
```
Comment (rakuco) [inline recipes-browser/chromium/chromium.inc]: The license changes I have here look larger than what's in this diff:

```diff
 LIC_FILES_CHKSUM = "\
     file://${S}/LICENSE;md5=0fca02217a5d49a14dfe2d11837bb34d \
     file://${S}/LICENSE.chromium_os;md5=87dd8458232da630f5617873d42d8350 \
@@ -92,6 +57,7 @@
     file://${S}/third_party/android_deps/libs/androidx_asynclayoutinflater_asynclayoutinflater/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_cardview_cardview/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_collection_collection/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+    file://${S}/third_party/android_deps/libs/androidx_concurrent_concurrent_futures/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_coordinatorlayout_coordinatorlayout/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_core_core/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_cursoradapter_cursoradapter/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
@@ -118,7 +84,6 @@
     file://${S}/third_party/android_deps/libs/androidx_localbroadcastmanager_localbroadcastmanager/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_media_media/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_mediarouter_mediarouter/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
-    file://${S}/third_party/android_deps/libs/androidx_multidex_multidex/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_palette_palette/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_preference_preference/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/androidx_print_print/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
@@ -189,7 +154,10 @@
     file://${S}/third_party/android_deps/libs/com_google_android_gms_play_services_vision/LICENSE;md5=32c019d8c1f8222a8aad9c3e1f1d2d10 \
     file://${S}/third_party/android_deps/libs/com_google_android_gms_play_services_vision_common/LICENSE;md5=32c019d8c1f8222a8aad9c3e1f1d2d10 \
     file://${S}/third_party/android_deps/libs/com_google_android_material_material/LICENSE;md5=7f7d74108ee1b7a743cca7d9a86784d6 \
+    file://${S}/third_party/android_deps/libs/com_google_dagger_dagger/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
+    file://${S}/third_party/android_deps/libs/com_google_guava_listenablefuture/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_deps/libs/com_google_protobuf_protobuf_lite/LICENSE;md5=37b5762e07f0af8c74ce80a8bda4266b \
+    file://${S}/third_party/android_deps/libs/javax_inject_javax_inject/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_media/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/android_opengl/LICENSE;md5=d10e92761a860d4113a7a525c78daf13 \
     file://${S}/third_party/android_sdk/LICENSE;md5=a9559ed17808a8b10eec6672f993ce75 \
@@ -202,7 +170,7 @@
     file://${S}/third_party/arcore-android-sdk/LICENSE;md5=c1bc4f7fbdd7a5a3f2c0dfefd457665e \
     file://${S}/third_party/ashmem/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/axe-core/LICENSE;md5=f27a50d2e878867827842f2c60e30bfc \
-    file://${S}/third_party/boringssl/src/LICENSE;md5=e91719b7727a2d3464ec006154275572 \
+    file://${S}/third_party/boringssl/src/LICENSE;md5=9b082148f9953258347788edb83e401b \
     file://${S}/third_party/boringssl/src/third_party/fiat/LICENSE;md5=97e5220ace241cda4cb42596a7d1a82e \
     file://${S}/third_party/breakpad/LICENSE;md5=828152b41153892a725f0e7f56faff73 \
     file://${S}/third_party/brotli/LICENSE;md5=941ee9cd1609382f946352712a319b4b \
@@ -222,11 +190,9 @@
     file://${S}/third_party/dom_distiller_js/LICENSE;md5=1399a5b89c7ed8167d261770432511e8 \
     file://${S}/third_party/emoji-segmenter/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/expat/src/expat/COPYING;md5=5b8620d98e49772d95fc1d291c26aa79 \
-    file://${S}/third_party/feed/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
-    file://${S}/third_party/ffmpeg/CREDITS.chromium;md5=f74504a10e06dc175c0e8e7e15105c05 \
+    file://${S}/third_party/ffmpeg/CREDITS.chromium;md5=5af3cfcd2117a0fe5d3b2f7b488c4b3b \
     file://${S}/third_party/flac/COPYING.Xiph;md5=a2c4b71c0198682376d483eb5bcc9197 \
     file://${S}/third_party/flatbuffers/LICENSE;md5=a873c5645c184d51e0f9b34e1d7cf559 \
-    file://${S}/third_party/flot/LICENSE.txt;md5=fe1ea2158d814e4fc7227c8c6ccd7ab3 \
     file://${S}/third_party/fontconfig/src/COPYING;md5=7a0449e9bc5370402a94c00204beca3d \
     file://${S}/third_party/freetype/src/docs/FTL.TXT;md5=9f37b4e6afa3fef9dba8932b16bd3f97 \
     file://${S}/third_party/gestures/LICENSE;md5=8a978a7211275297a3d816117be58ebf \
@@ -257,6 +223,7 @@
     file://${S}/third_party/libaddressinput/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/libcxx-pretty-printers/LICENSE;md5=d32239bcb673463ab874e80d47fae504 \
     file://${S}/third_party/libevdev/LICENSE;md5=9ee2ef12ddcc3b79de961c76918569b6 \
+    file://${S}/third_party/libgifcodec/LICENSE.md;md5=1018a09551825ebc78d78f84855e07fa \
     file://${S}/third_party/libipp/LICENSE;md5=af9e58383a1b2b17c75c6c9ff561ca9d \
     file://${S}/third_party/libjingle_xmpp/LICENSE;md5=ad296492125bc71530d06234d9bfebe0 \
     file://${S}/third_party/libjpeg/LICENSE;md5=b44b6615c3ed3bad7c25c9d66de81b05 \
@@ -281,7 +248,7 @@
     file://${S}/third_party/lottie/LICENSE;md5=827837648055a0bfb7782b91ab42eed1 \
     file://${S}/third_party/lzma_sdk/LICENSE;md5=a8d1f5a28cfd35e6dfea09b82672fae7 \
     file://${S}/third_party/material_design_icons/LICENSE;md5=175792518e4ac015ab6696d16c4f607e \
-    file://${S}/third_party/mesa_headers/LICENSE;md5=89b61de67f7dc5e1c536aa23192b1468 \
+    file://${S}/third_party/mesa_headers/LICENSE;md5=072744090020e72ad7387e201873fe43 \
     file://${S}/third_party/metrics_proto/LICENSE;md5=0fca02217a5d49a14dfe2d11837bb34d \
     file://${S}/third_party/microsoft_webauthn/LICENSE;md5=b98fddd052bb2f5ddbcdbd417ffb26a8 \
     file://${S}/third_party/minigbm/LICENSE;md5=fcd558ab2fd7f53408826e80eb0c11eb \
@@ -295,7 +262,7 @@
     file://${S}/third_party/openvr/src/LICENSE;md5=3f0ba432004f9cf11d8186503a8bd3d7 \
     file://${S}/third_party/opus/src/COPYING;md5=e304cdf74c2a1b0a33a5084c128a23a3 \
     file://${S}/third_party/ots/LICENSE;md5=e06eff2aa65b917034a81599bea73dc4 \
-    file://${S}/third_party/pdfium/LICENSE;md5=defd64fe3646eb60a5d01fcbcc8b6b7c \
+    file://${S}/third_party/pdfium/LICENSE;md5=efe815679251504905d0f2c88a77c886 \
     file://${S}/third_party/perfetto/NOTICE;md5=aeb863d080c67bc38075748338c57a7f \
     file://${S}/third_party/pffft/LICENSE;md5=0f39e43e9bc20e7e103e54750e1ec3a2 \
     file://${S}/third_party/ply/LICENSE;md5=f298aa3b0cfe0b0ea96acabf944529c0 \
@@ -309,7 +276,7 @@
     file://${S}/third_party/rnnoise/COPYING;md5=1890bf89a18f8339491894a0b45428bf \
     file://${S}/third_party/sfntly/COPYING.txt;md5=2326ae5ca22e299a528433bebb02131f \
     file://${S}/third_party/simplejson/LICENSE.txt;md5=8a9f8dcfcd28c34daa443e5b1d29813b \
-    file://${S}/third_party/skia/LICENSE;md5=d25bb58a1be2e1af9b58d31565a206dc \
+    file://${S}/third_party/skia/LICENSE;md5=822f02cc7736281816581cd064afbb1c \
     file://${S}/third_party/smhasher/LICENSE;md5=0d948322ab524e2b74f55eb8ef57c74a \
     file://${S}/third_party/snappy/src/COPYING;md5=f62f3080324a97b3159a7a7e61812d0c \
     file://${S}/third_party/speech-dispatcher/COPYING;md5=5828ebb4b7590269a994ed46b0a70df0 \
@@ -331,6 +298,7 @@
     file://${S}/third_party/webrtc/LICENSE;md5=ad296492125bc71530d06234d9bfebe0 \
     file://${S}/third_party/woff2/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57 \
     file://${S}/third_party/wtl/Ms-PL.txt;md5=1d2c3cdd7e748a6bc4213321d9157c5a \
+    file://${S}/third_party/wuffs/LICENSE;md5=2ee41112a44fe7014dce33e26468ba93 \
     file://${S}/third_party/xdg-utils/LICENSE;md5=a5367a90934098d6b05af3b746405014 \
     file://${S}/third_party/yasm/source/patched-yasm/COPYING;md5=a12d8903508fb6bfd49d8d82c6170dd9 \
     file://${S}/third_party/zlib/LICENSE;md5=f09575dbfb09420642318b413159496f \
```

some Android deps in third_party do not have appropriate license information and were left out so I'm tempted to just ignore them, and there's `third_party/devtools-frontend` that is licensed under the 3-clause BSD License but does not have a README.chromium in place. For the record, I've filed [a bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1053941) for the latter.

Comment (msisov) [reply recipes-browser/chromium/chromium.inc]: I rely on the bitbake and update only those licences that it complains about.

Comment (rakuco) [reply recipes-browser/chromium/chromium.inc]: bitbake only complains about existing files whose hashes changed or which no longer exist, so relying on means that no new directories will ever be added to the list AFAICS.

Comment (msisov) [reply recipes-browser/chromium/chromium.inc]: Do you want me to go through all the news deps manually? or is there a tool for that?

Comment (rakuco) [reply recipes-browser/chromium/chromium.inc]: In this case I think you can just use the diff above and check what the new license files are. In general, I use [this script](https://gist.github.com/rakuco/dd124664be0a016d7742a4e0597c6169) that I never got the chance to finish and add to the repository :-)

Comment (msisov) [reply recipes-browser/chromium/chromium.inc]: done. One remark - androidx_multidex_multidex wasn't removed. It is still the same.

**Comment by msisov** [comment]
> lgtm, thanks. Please just make sure all the information you added to the PR message (the license information, for example) is also included in the commit message for posterity.

done

**Pull request merged**